doc/user/client.md

```diff
@@ -0,0 +1,252 @@
+	labelSelector := labels.SelectorFromSet(labelsForApp(app.Name))
+	listOps := &client.ListOptions{Namespace: app.Namespace, LabelSelector: labelSelector}
+	if err = r.client.List(context.TODO(), listOps, podList); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// Update status.Nodes if needed.
+	podNames := getPodNames(podList.Items)
+	if !reflect.DeepEqual(podNames, app.Status.Nodes) {
+		app.Status.Nodes = podNames
+		if err := r.client.Update(context.TODO(), app); err != nil {
+			return reconcile.Result{}, err
+		}
+	}
+
+	return reconcile.Result{}, nil
+}
+
+// deploymentForApp returns a app Deployment object.
+func (r *ReconcileMemcached) deploymentForApp(m *appv1alpha1.App) *appsv1.Deployment {
+	lbls := labelsForApp(m.Name)
+	replicas := m.Spec.Size
+
+	dep := &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "apps/v1",
+			Kind:       "Deployment",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      m.Name,
+			Namespace: m.Namespace,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: lbls,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: lbls,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{{
+						Image:   "app:alpine",
+						Name:    "app",
+						Command: []string{"app", "-a=64", "-b"},
+						Ports: []corev1.ContainerPort{{
+							ContainerPort: 10000,
+							Name:          "app",
+						}},
+					}},
+				},
+			},
+		},
+	}
+
+	// Set App instance as the owner and controller.
+	// NOTE: calling SetControllerReference, and setting owner references in
+	// general, is important as it allows deleted objects to be garbage collected.
+	controllerutil.SetControllerReference(m, dep, r.scheme)
+	return dep
+}
+
+// labelsForApp creates a simple set of labels for App.
+func labelsForApp(name string) map[string]string {
+	return map[string]string{"app_name": "app", "app_cr": name}
+}
+```
+
+[repo-controller-runtime]:https://github.com/kubernetes-sigs/controller-runtime
+[org-kubernetes-sigs]:https://github.com/kubernetes-sigs
+[site-kubernetes]:https://kubernetes.io/
+[site-kubernetes-controllers]:https://kubernetes.io/docs/concepts/workloads/controllers/
+[site-operators]:https://coreos.com/blog/introducing-operator-framework
+[doc-split-client]:https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/client/split.go#L26-L28
+[doc-rest-config]:https://godoc.org/k8s.io/client-go/rest#Config
+[doc-client-constr]:https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/client/client.go#L44
+[doc-client-fake]:https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/client/fake/client.go
+[doc-client-options]:https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/client/client.go#L35
+[doc-reconcile-reconciler]:https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/reconcile/reconcile.go#L35
```

Comment:
For some of the links, I think it might be better to point to the [kubebuilder book](https://book.kubebuilder.io/) or the [godoc](https://godoc.org/sigs.k8s.io/controller-runtime/pkg). These might be a better place to point users to get info than the code IMO

Reply:
The runtime godoc might be better for each pkg. That's more readable.